### DISCUSSION TITLE
Does taskfile support LSF or GRD

### DISCUSSION THREAD

**Discussion by texan4ever** [go-task/task#654] [Q&A]
I am wondering if taskfile has support to send jobs to an LSF or GIRD farm?

If not, can anyone suggest programs similar to taskfile that do support LSF or GRID.  (I hate writing makefiles)

**Comment by andreynering** [comment]
Hi @texan4ever,

Sorry, but I don't know what "LSF or GIRD farm" is.

**Comment by texan4ever** [comment]
LSF or GRID are software packages that manage jobs on a CPU farm (like at a university or corporate environment)
The purpose is or LSF or GRID is to make much more effective use of CPU resources instead of direct login or console usage.

Job submissions require at least the following information from the user:
- the number of CPU's required
- maximum RAM required
- shell script to be run

GRID uses a qsub command and LSF uses a bsub command. The user would need to tell taskfile the minimum following information:
- name of job submission command
- options for job submission
So a version of taskfile that worked with LSF/GRD would submit jobs to the farm instead of running jobs locally. While the jobs were running on the farm taskfile would have to monitor the status of dependencies while it waited for the job to finish.

An introduction to LSF:
https://www.ibm.com/docs/en/spectrum-lsf/10.1.0?topic=overview-lsf-introduction

FYI: LSF and GRID (also called UGE) do the same thing.  LSF is sold by IBM, GRID sold by Oracle.

**Reply by andreynering** [reply]
Hi @texan4ever,

Sorry, but this sounds really out-of-the-scope of Task. We don't plan to support anything similar. Tasks will always run locally.

The closer we can reach at some point is running tasks inside Docker containers.
